python-libraries/data-platform-catalogue/data_platform_catalogue/client.py

```diff
@@ -183,6 +190,15 @@ def _create_or_update_entity(self, data) -> str:
 
         return response.dict()["fullyQualifiedName"]
 
+    def get_user_id(self, user_email: str):
+        """
+        returns the user id from openmetadata when given a user's email
+        """
+        username = user_email.split("@")[0]
+        user_id = self.metadata.get_by_name(entity=User, fqn=username).id
```

Comment:
We'll need to make sure we set up the name in the same way when we do the control panel bit that provisions users the first time they log into the platform.

I can see this potentially going wrong if people mix up justice emails and other email domains that are in use across the department. Even if people have both they could be in a different format. I wonder if we should just enforce that a user email always has to end in justice.gov.uk?

Reply:
The username in openmetadata is aways what ever comes before the `@` but we are using azure to authenticate so it'll always be justice.gov.uk (i think anyway), and we can't be sure peoples names will always follow the same spelling or format. Maybe we should enforce all data providers to use an @justice.gov.uk email, but one to raise for further discussion

Reply:
Yeah, let's run with this assumption for now 👍🏻